sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/PartitionOwnership.cs

```diff
@@ -76,21 +76,21 @@ public class PartitionOwnership
         ///
         /// <param name="eventHubName">The name of the specific Event Hub this partition ownership is associated with, relative to the Event Hubs namespace that contains it.</param>
         /// <param name="consumerGroup">The name of the consumer group this partition ownership is associated with.</param>
-        /// <param name="ownerIdentifier">The identifier of the associated <see cref="EventProcessor" /> instance.</param>
+        /// <param name="ownerIdentifier">The identifier of the associated <see cref="EventProcessor{T}" /> instance.</param>
         /// <param name="partitionId">The identifier of the Event Hub partition this partition ownership is associated with.</param>
-        /// <param name="offset">The offset of the last <see cref="EventData" /> received by the associated <see cref="IPartitionProcessor" />.</param>
-        /// <param name="sequenceNumber">The sequence number of the last <see cref="EventData" /> received by the associated <see cref="IPartitionProcessor" />.</param>
+        /// <param name="offset">The offset of the last <see cref="EventData" /> received by the associated partition processor.</param>
+        /// <param name="sequenceNumber">The sequence number of the last <see cref="EventData" /> received by the associated partition processor.</param>
         /// <param name="lastModifiedTime">The date and time, in UTC, that the last update was made to this ownership.</param>
         /// <param name="eTag">The entity tag needed to update this ownership.</param>
         ///
-        internal PartitionOwnership(string eventHubName,
-                                    string consumerGroup,
-                                    string ownerIdentifier,
-                                    string partitionId,
-                                    long? offset = null,
-                                    long? sequenceNumber = null,
-                                    DateTimeOffset? lastModifiedTime = null,
-                                    string eTag = null)
+        protected internal PartitionOwnership(string eventHubName,
```

Comment:
This is a bit odd.   Only created by a limited scope, but open for changing properties to `public`.    Was that intentional?

Reply:
Yes. The main idea was not letting the users instantiate `PartitionOwnership` (unless for mocking purposes) and making some information immutable (`PartitionId`, `EventHubName`, `ConsumerGroup` and `OwnerIdentifier`). However, some properties need to be changed (`Offset`, `SequenceNumber` and `LastModifiedTime` upon checkpointing, `ETag` upon claim). Do you think there's a better way of doing that?

Reply:
Nope, I think you've got the right approach for your scenario.  I don't have a reason against.   It just looked a bit inconsistent and I wasn't sure if there was some refactoring that got forgotten.